### PR TITLE
fix: Avoided dereferencing NULL pointer

### PR DESCRIPTION
**Pull request by Ald392**
issue : https://github.com/milvus-io/milvus/issues/34835

### CODE REVIEW

File: internal/core/src/indexbuilder/init_c.cpp [hunk 1]
```diff
@@ -23,6 +24,9 @@ char*
 IndexBuilderSetSimdType(const char* value) {
     auto real_type = milvus::config::KnowhereSetSimdType(value);
     char* ret = reinterpret_cast<char*>(malloc(real_type.length() + 1));
+    if (ret == nullptr) {
+        throw std::runtime_error("Error: Memory allocation failed");
```
Comment (longjiquan) [inline internal/core/src/indexbuilder/init_c.cpp:28]: Better to use "AssertInfo" instead.

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: internal/core/src/indexbuilder/init_c.cpp (hunk 1 above)
Comment (Ald392) [reply internal/core/src/indexbuilder/init_c.cpp:28]: Thanks for the suggestion. Currently working on it.

**Review by longjiquan — COMMENTED** [review]
/lgtm

**Comment by longjiquan** [comment]
rerun ut

**Comment by Ald392** [comment]
@longjiquan how should we proceed ?

**Pull request merged**